registry/storage/driver/hdfsweb/fs.go

```diff
@@ -0,0 +1,452 @@
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net"
+	"net/http"
+	"net/url"
+	"strconv"
+	"time"
+)
+
+const (
+	webHdfsVer string = "/webhdfs/v1"
+)
+
+// FileSystem provides the implementation of sending http requests to hdfs server,
+// and the parameters to build the http url.
+type FileSystem struct {
+	NameNodeHost string
+	NameNodePort string
+	Username     string
+	BlockSize    uint64
+	BufferSize   int
+	Replication  uint16
+}
+
+// Create creates a new file and stores its content in HDFS.
+func (fs *FileSystem) Create(
+	reader io.Reader,
+	path string) error {
+
+	transport := http.Transport{Dial: dialTimeout}
+	client := http.Client{
+		Transport: &transport,
+	}
+
+	v := url.Values{
+		"user.name":   []string{fs.Username},
+		"op":          []string{"CREATE"},
+		"overwrite":   []string{"true"},
+		"blocksize":   []string{strconv.FormatInt(int64(fs.BlockSize), 10)},
+		"replication": []string{strconv.FormatInt(int64(fs.Replication), 10)},
+		"permission":  []string{"0666"},
+		"buffersize":  []string{strconv.FormatInt(int64(fs.BufferSize), 10)},
+	}
+
+	u, err := fs.buildRequestURL(path, v)
+	if err != nil {
+		return err
+	}
+
+	// take over default transport to avoid redirect
+	tr := &http.Transport{}
+	req, err := http.NewRequest("PUT", u.String(), nil)
+	if err != nil {
+		return err
+	}
+	tripResp, err := tr.RoundTrip(req)
+	if err != nil {
+		return err
+	}
+	defer tripResp.Body.Close()
+	// extract returned url in header.
+	loc := tripResp.Header.Get("Location")
+	u, err = url.ParseRequestURI(loc)
+	if err != nil {
+		return err
+	}
+
+	req, err = http.NewRequest("PUT", u.String(), reader)
+	if err != nil {
+		return err
+	}
+	resp, err := client.Do(req)
+	if err != nil {
+		return err
+	}
+	defer resp.Body.Close()
+	if resp.StatusCode != http.StatusCreated {
+		body, err := ioutil.ReadAll(resp.Body)
+		if err != nil {
+			return err
+		}
+		exp := RespException{}
+		if err := json.Unmarshal(body, &exp); err != nil {
+			return err
+		} else if exp.RemoteException.Exception != "" {
+			return exp.RemoteException
+		} else {
+			return fmt.Errorf("[Create] path(%s) unknown error return (%d)", path, resp.StatusCode)
+		}
+	}
+	return nil
+}
+
+//Open opens the specificed Path and returns its content to be accessed locally.
+func (fs *FileSystem) Open(path string, offset, length int64) (io.ReadCloser, error) {
+	transport := http.Transport{Dial: dialTimeout}
+	client := http.Client{
+		Transport: &transport,
+	}
+
+	v := url.Values{
+		"user.name":  []string{fs.Username},
+		"op":         []string{"OPEN"},
+		"offset":     []string{strconv.FormatInt(offset, 10)},
+		"length":     []string{strconv.FormatInt(length, 10)},
+		"buffersize": []string{strconv.FormatInt(int64(fs.BufferSize), 10)},
+	}
+
+	u, err := fs.buildRequestURL(path, v)
+	if err != nil {
+		return nil, err
+	}
+
+	req, err := http.NewRequest("GET", u.String(), nil)
+	if err != nil {
+		return nil, err
+	}
+	resp, err := client.Do(req)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
I re-defines the client as,
```go
client := http.Client{
        Transport: &http.Transport{Dial: dialTimeout},
}
```
So, it seems not to be proper to use `http.Get` that involves default transport here, right?